# **PROJETO ANÁLISE ESTATÍSTICA DO ENEM 2019**

Este projeto analisa os resultados do ENEM no estado de São Paulo do ano de 2019.

Os dados foram extraídos do site do INEP:

http://inep.gov.br/microdados

In [5]:
import numpy as np
import pandas as pd

In [6]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [7]:
dados = pd.read_csv('/content/drive/MyDrive/curso_estatistica_python/Scripts/microdados_enem_2019_sp.csv',sep=';', encoding='iso-8859-1')
# encoding: codificação de caracteres, normalmente utiliza-se o iso-8859-1, utf-8, latin-1)

## Extração e Limpeza dos dados

In [8]:
pd.options.display.max_columns = 44
#comando de ajuste do número máximo de colunas para 44

In [9]:
dados.head()
#visualisa e retorna as cinco primeiras linhas do DataFrame

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA NO_MUNICIPIO_RESIDENCIA  \
0     1,90E+11                  3546801            Santa Isabel   
1     1,90E+11                  3538204             Pinhalzinho   
2     1,90E+11                  3550308               São Paulo   
3     1,90E+11                  3548708   São Bernardo do Campo   
4     1,90E+11                  3549409    São Joaquim da Barra   

   CO_UF_RESIDENCIA SG_UF_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0                35               SP        22       M                1   
1                35               SP        19       M                1   
2                35               SP        17       F                1   
3                35               SP        19       M                1   
4                35               SP        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  CO_MUNICIPIO_NASCIMENTO  \
0            3                 1                1506500.0   
1            3                 1                1503804.0   
2            3                 1                1501402.0   
3            3                 1                3548708.0   
4            1                 1                3549409.0   

  NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO SG_UF_NASCIMENTO  TP_ST_CONCLUSAO  \
0    Santa Izabel do Pará              15.0               PA                1   
1                 Jacundá              15.0               PA                1   
2                   Belém              15.0               PA                2   
3   São Bernardo do Campo              35.0               SP                2   
4    São Joaquim da Barra              35.0               SP                1   

   TP_ANO_CONCLUIU  TP_ESCOLA  TP_ENSINO  IN_TREINEIRO   CO_ESCOLA  \
0                4          1        1.0             0         NaN   
1                2          1        1.0             0         NaN   
2                0          3        1.0             0  15567761.0   
3                0          2        1.0             0  35904958.0   
4                3          1        1.0             0         NaN   

   CO_MUNICIPIO_ESC       NO_MUNICIPIO_ESC  CO_UF_ESC SG_UF_ESC  \
0               NaN                    NaN        NaN       NaN   
1               NaN                    NaN        NaN       NaN   
2         1501402.0                  Belém       15.0        PA   
3         3548708.0  São Bernardo do Campo       35.0        SP   
4               NaN                    NaN        NaN       NaN   

   TP_DEPENDENCIA_ADM_ESC  TP_LOCALIZACAO_ESC  TP_SIT_FUNC_ESC  \
0                     NaN                 NaN              NaN   
1                     NaN                 NaN              NaN   
2                     4.0                 1.0              1.0   
3                     2.0                 1.0              1.0   
4                     NaN                 NaN              NaN   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  \
0               1               1               1               1      5646.0   
1               0               0               0               0         NaN   
2               1               1               1               1      5762.0   
3               1               1               1               1      4022.0   
4               0               0               0               0         NaN   

   NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  \
0      5858.0      5929.0      6002.0          0                1.0   
1         NaN         NaN         NaN          1                NaN   
2      6411.0      6343.0      7318.0          0                1.0   
3      4248.0      5096.0      3932.0          1                1.0   
4         NaN         NaN         NaN          1                NaN   

   NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  \
0          160.0          200.0          180.0          200.0          200.0   

In [10]:
dados.shape

(813772, 44)

In [11]:
# Análise dos tipos de atributos.
# object: strings
# int64: inteiros
# float64: reais
# complex: complexos
dados.dtypes

NU_INSCRICAO                object
CO_MUNICIPIO_RESIDENCIA      int64
NO_MUNICIPIO_RESIDENCIA     object
CO_UF_RESIDENCIA             int64
SG_UF_RESIDENCIA            object
NU_IDADE                     int64
TP_SEXO                     object
TP_ESTADO_CIVIL              int64
TP_COR_RACA                  int64
TP_NACIONALIDADE             int64
CO_MUNICIPIO_NASCIMENTO    float64
NO_MUNICIPIO_NASCIMENTO     object
CO_UF_NASCIMENTO           float64
SG_UF_NASCIMENTO            object
TP_ST_CONCLUSAO              int64
TP_ANO_CONCLUIU              int64
TP_ESCOLA                    int64
TP_ENSINO                  float64
IN_TREINEIRO                 int64
CO_ESCOLA                  float64
CO_MUNICIPIO_ESC           float64
NO_MUNICIPIO_ESC            object
CO_UF_ESC                  float64
SG_UF_ESC                   object
TP_DEPENDENCIA_ADM_ESC     float64
TP_LOCALIZACAO_ESC         float64
TP_SIT_FUNC_ESC            float64
TP_PRESENCA_CN               int64
TP_PRESENCA_CH               int64
TP_PRESENCA_LC               int64
TP_PRESENCA_MT               int64
NU_NOTA_CN                 float64
NU_NOTA_CH                 float64
NU_NOTA_LC                 float64
NU_NOTA_MT                 float64
TP_LINGUA                    int64
TP_STATUS_REDACAO          float64
NU_NOTA_COMP1              float64
NU_NOTA_COMP2              float64
NU_NOTA_COMP3              float64
NU_NOTA_COMP4              float64
NU_NOTA_COMP5              float64
NU_NOTA_REDACAO            float64
Q025                        object
dtype: object

In [12]:
dados1 = dados.drop(columns=['CO_MUNICIPIO_RESIDENCIA'])
# Foi removido a coluna do código do município de residência por ser repetitivo com a coluna nome do município.

In [13]:
dados1 = dados1.drop(columns=['CO_UF_RESIDENCIA','SG_UF_RESIDENCIA','CO_MUNICIPIO_NASCIMENTO'],)
#Não sendo necessário, pois a análise está sendo feita apenas com o estado de SP

In [14]:
dados1.head(40)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0      1,90E+11            Santa Isabel        22       M                1   
1      1,90E+11             Pinhalzinho        19       M                1   
2      1,90E+11               São Paulo        17       F                1   
3      1,90E+11   São Bernardo do Campo        19       M                1   
4      1,90E+11    São Joaquim da Barra        37       M                2   
5      1,90E+11              Embu-Guaçu        17       F                1   
6      1,90E+11     São José dos Campos        28       F                1   
7      1,90E+11     São José dos Campos        25       F                2   
8      1,90E+11               Guarulhos        23       F                1   
9      1,90E+11               São Paulo        19       F                1   
10     1,90E+11   São José do Rio Preto        18       F                1   
11     1,90E+11                  Sumaré        22       F                1   
12     1,90E+11                  Franca        32       M                1   
13     1,90E+11               São Paulo        19       F                1   
14     1,90E+11                   Bauru        18       M                1   
15     1,90E+11             Santo André        20       F                1   
16     1,90E+11                Campinas        20       F                1   
17     1,90E+11               São Paulo        26       M                1   
18     1,90E+11             Santo André        44       F                2   
19     1,90E+11               São Paulo        42       F                1   
20     1,90E+11                  Santos        15       F                1   
21     1,90E+11                Panorama        16       F                1   
22     1,90E+11              Jaguariúna        19       M                1   
23     1,90E+11               São Paulo        17       F                1   
24     1,90E+11   São Bernardo do Campo        32       F                1   
25     1,90E+11                     Poá        57       M                2   
26     1,90E+11             Carapicuíba        21       M                1   
27     1,90E+11                   Guará        17       F                1   
28     1,90E+11                 Cajamar        17       F                1   
29     1,90E+11               São Paulo        50       F                1   
30     1,90E+11                  Osasco        17       F                1   
31     1,90E+11               São Paulo        19       F                1   
32     1,90E+11               São Paulo        19       M                1   
33     1,90E+11               São Paulo        38       F                2   
34     1,90E+11               São Paulo        50       M                3   
35     1,90E+11               São Paulo        18       F                1   
36     1,90E+11                Campinas        17       F                1   
37     1,90E+11               São Paulo        18       F                1   
38     1,90E+11     São José dos Campos        17       M                1   
39     1,90E+11                  Osasco        38       F                1   

    TP_COR_RACA  TP_NACIONALIDADE NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO  \
0             3                 1    Santa Izabel do Pará              15.0   
1             3                 1                 Jacundá              15.0   
2             3                 1                   Belém              15.0   
3             3                 1   São Bernardo do Campo              35.0   
4             1                 1    São Joaquim da Barra              35.0   
5             3                 1             Itacoatiara              13.0   
6             1                 1                  Manaus              13.0   
7             1                 1     São José dos Campos              35.0   
8             1                 1               São Paulo              35.0   
9             

In [15]:
dados1 = dados1.drop(columns=['NO_MUNICIPIO_NASCIMENTO','CO_UF_NASCIMENTO','SG_UF_NASCIMENTO',
                              'TP_ANO_CONCLUIU','TP_ENSINO','CO_MUNICIPIO_ESC','CO_UF_ESC','SG_UF_ESC'],)

In [16]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

    CO_ESCOLA       NO_MUNICIPIO_ESC  TP_DEPENDENCIA_ADM_ESC  \
0         NaN                    NaN                     NaN   
1         NaN                    NaN                     NaN   
2  15567761.0                  Belém                     4.0   
3  35904958.0  São Bernardo do Campo                     2.0   
4         NaN                    NaN                     NaN   

   TP_LOCALIZACAO_ESC  TP_SIT_FUNC_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0                 NaN              NaN               1               1   
1                 NaN              NaN               0               0   
2                 1.0              1.0               1               1   
3                 1.0              1.0               1               1   
4                 NaN              NaN               0               0   

   TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0               1               1      5646.0      5858.0      5929.0   
1               0               0         NaN         NaN         NaN   
2               1               1      5762.0      6411.0      6343.0   
3               1               1      4022.0      4248.0      5096.0   
4               0               0         NaN         NaN         NaN   

   NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0      6002.0          0                1.0          160.0          200.0   
1         NaN          1                NaN            NaN            NaN   
2      7318.0          0                1.0          160.0          120.0   
3      3932.0          1                1.0           80.0          160.0   
4         NaN          1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO Q025  
0          180.0          200.0          200.0            940.0    B  
1            NaN            NaN            NaN              NaN    A  
2          200.0          200.0          200.0            880.0    A  
3          120.0          100.0          100.0            560.0    B  
4            NaN            NaN            NaN              NaN    B

In [17]:
dados1 = dados1.drop(columns=['TP_DEPENDENCIA_ADM_ESC','TP_LOCALIZACAO_ESC','TP_SIT_FUNC_ESC'],)

In [18]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

    CO_ESCOLA       NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0         NaN                    NaN               1               1   
1         NaN                    NaN               0               0   
2  15567761.0                  Belém               1               1   
3  35904958.0  São Bernardo do Campo               1               1   
4         NaN                    NaN               0               0   

   TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0               1               1      5646.0      5858.0      5929.0   
1               0               0         NaN         NaN         NaN   
2               1               1      5762.0      6411.0      6343.0   
3               1               1      4022.0      4248.0      5096.0   
4               0               0         NaN         NaN         NaN   

   NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0      6002.0          0                1.0          160.0          200.0   
1         NaN          1                NaN            NaN            NaN   
2      7318.0          0                1.0          160.0          120.0   
3      3932.0          1                1.0           80.0          160.0   
4         NaN          1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO Q025  
0          180.0          200.0          200.0            940.0    B  
1            NaN            NaN            NaN              NaN    A  
2          200.0          200.0          200.0            880.0    A  
3          120.0          100.0          100.0            560.0    B  
4            NaN            NaN            NaN              NaN    B

In [19]:
dados1.shape

(813772, 29)

Corrigindo erros nas notas

In [20]:
dados1.loc[:,'NU_NOTA_CN'] /= 10
#Nota com casas decimais erradas. Dividindo por 10 aproxima ao real resultado

In [21]:
dados1.head(12)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0      1,90E+11            Santa Isabel        22       M                1   
1      1,90E+11             Pinhalzinho        19       M                1   
2      1,90E+11               São Paulo        17       F                1   
3      1,90E+11   São Bernardo do Campo        19       M                1   
4      1,90E+11    São Joaquim da Barra        37       M                2   
5      1,90E+11              Embu-Guaçu        17       F                1   
6      1,90E+11     São José dos Campos        28       F                1   
7      1,90E+11     São José dos Campos        25       F                2   
8      1,90E+11               Guarulhos        23       F                1   
9      1,90E+11               São Paulo        19       F                1   
10     1,90E+11   São José do Rio Preto        18       F                1   
11     1,90E+11                  Sumaré        22       F                1   

    TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0             3                 1                1          1             0   
1             3                 1                1          1             0   
2             3                 1                2          3             0   
3             3                 1                2          2             0   
4             1                 1                1          1             0   
5             3                 1                2          2             0   
6             1                 1                1          1             0   
7             1                 1                1          1             0   
8             1                 1                1          1             0   
9             3                 1                1          1             0   
10            1                 1                2          2             0   
11            4                 1                1          1             0   

     CO_ESCOLA       NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0          NaN                    NaN               1               1   
1          NaN                    NaN               0               0   
2   15567761.0                  Belém               1               1   
3   35904958.0  São Bernardo do Campo               1               1   
4          NaN                    NaN               0               0   
5          NaN                    NaN               1               1   
6          NaN                    NaN               0               0   
7          NaN                    NaN               0               0   
8          NaN                    NaN               1               1   
9          NaN                    NaN               0               0   
10  35028927.0  São José do Rio Preto               1               1   
11         NaN                    NaN               0               0   

    TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0                1               1       564.6      5858.0      5929.0   
1                0               0         NaN         NaN         NaN   
2                1               1       576.2      6411.0      6343.0   
3                1               1       402.2      4248.0      5096.0   
4                0               0         NaN         NaN         NaN   
5                1               1       399.6      4268.0      4872.0   
6                0               0         NaN         NaN         NaN   
7                0               0         NaN         NaN         NaN   
8                1               1       625.7      6465.0      6293.0   
9                0               0         NaN         NaN         NaN   
10               1               1       449.5      5333.0      5559.0   
11               0               0         NaN         NaN         NaN   

    NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  N

In [22]:
dados1.loc[:,'NU_NOTA_CH'] /= 10
dados1.loc[:,'NU_NOTA_LC'] /= 10
dados1.loc[:,'NU_NOTA_MT'] /= 10

In [23]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

    CO_ESCOLA       NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0         NaN                    NaN               1               1   
1         NaN                    NaN               0               0   
2  15567761.0                  Belém               1               1   
3  35904958.0  São Bernardo do Campo               1               1   
4         NaN                    NaN               0               0   

   TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0               1               1       564.6       585.8       592.9   
1               0               0         NaN         NaN         NaN   
2               1               1       576.2       641.1       634.3   
3               1               1       402.2       424.8       509.6   
4               0               0         NaN         NaN         NaN   

   NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0       600.2          0                1.0          160.0          200.0   
1         NaN          1                NaN            NaN            NaN   
2       731.8          0                1.0          160.0          120.0   
3       393.2          1                1.0           80.0          160.0   
4         NaN          1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO Q025  
0          180.0          200.0          200.0            940.0    B  
1            NaN            NaN            NaN              NaN    A  
2          200.0          200.0          200.0            880.0    A  
3          120.0          100.0          100.0            560.0    B  
4            NaN            NaN            NaN              NaN    B

Renomeando Variáveis (colunas)

In [24]:
dados1 = dados1.rename(columns={'NU_NOTA_REDACAO': 'NOTA_REDACAO'})

In [25]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

    CO_ESCOLA       NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0         NaN                    NaN               1               1   
1         NaN                    NaN               0               0   
2  15567761.0                  Belém               1               1   
3  35904958.0  São Bernardo do Campo               1               1   
4         NaN                    NaN               0               0   

   TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0               1               1       564.6       585.8       592.9   
1               0               0         NaN         NaN         NaN   
2               1               1       576.2       641.1       634.3   
3               1               1       402.2       424.8       509.6   
4               0               0         NaN         NaN         NaN   

   NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0       600.2          0                1.0          160.0          200.0   
1         NaN          1                NaN            NaN            NaN   
2       731.8          0                1.0          160.0          120.0   
3       393.2          1                1.0           80.0          160.0   
4         NaN          1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NOTA_REDACAO Q025  
0          180.0          200.0          200.0         940.0    B  
1            NaN            NaN            NaN           NaN    A  
2          200.0          200.0          200.0         880.0    A  
3          120.0          100.0          100.0         560.0    B  
4            NaN            NaN            NaN           NaN    B

In [26]:
dados1 = dados1.rename(columns={'NU_NOTA_CN': 'NOTA_CN','NU_NOTA_CH': 'NOTA_CH',
                                'NU_NOTA_LC': 'NOTA_LC','NU_NOTA_MT': 'NOTA_MT'})

In [27]:
dados1 = dados1.rename(columns={'NU_NOTA_COMP1': 'COMP1','NU_NOTA_COMP2': 'COMP2',
                                'NU_NOTA_COMP3': 'COMP3','NU_NOTA_COMP4': 'COMP4',
                                'NU_NOTA_COMP5':'COMP5'})

In [28]:
dados1 = dados1.rename(columns={'NU_IDADE': 'IDADE','TP_SEXO': 'SEXO',
                                'TP_COR_RACA': 'RACA','Q025': 'INTERNET',
                                'TP_ESCOLA':'ESCOLA'})

Renomeando Registros (Linhas)

In [29]:
dados1["RACA"] = dados1["RACA"].replace({0:"nao_declarado", 1:"branca", 2:"preta", 3:"parda", 4:"amarela", 5:"indigena"})
#De acordo com a numeração escolhida para a declaração de cor, foi substituída pelo nome em si para melhor análise

In [30]:
dados1.head(20)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL     RACA  \
0      1,90E+11            Santa Isabel     22    M                1    parda   
1      1,90E+11             Pinhalzinho     19    M                1    parda   
2      1,90E+11               São Paulo     17    F                1    parda   
3      1,90E+11   São Bernardo do Campo     19    M                1    parda   
4      1,90E+11    São Joaquim da Barra     37    M                2   branca   
5      1,90E+11              Embu-Guaçu     17    F                1    parda   
6      1,90E+11     São José dos Campos     28    F                1   branca   
7      1,90E+11     São José dos Campos     25    F                2   branca   
8      1,90E+11               Guarulhos     23    F                1   branca   
9      1,90E+11               São Paulo     19    F                1    parda   
10     1,90E+11   São José do Rio Preto     18    F                1   branca   
11     1,90E+11                  Sumaré     22    F                1  amarela   
12     1,90E+11                  Franca     32    M                1    parda   
13     1,90E+11               São Paulo     19    F                1   branca   
14     1,90E+11                   Bauru     18    M                1    parda   
15     1,90E+11             Santo André     20    F                1    parda   
16     1,90E+11                Campinas     20    F                1  amarela   
17     1,90E+11               São Paulo     26    M                1   branca   
18     1,90E+11             Santo André     44    F                2    parda   
19     1,90E+11               São Paulo     42    F                1    parda   

    TP_NACIONALIDADE  TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0                  1                1       1             0         NaN   
1                  1                1       1             0         NaN   
2                  1                2       3             0  15567761.0   
3                  1                2       2             0  35904958.0   
4                  1                1       1             0         NaN   
5                  1                2       2             0         NaN   
6                  1                1       1             0         NaN   
7                  1                1       1             0         NaN   
8                  1                1       1             0         NaN   
9                  1                1       1             0         NaN   
10                 1                2       2             0  35028927.0   
11                 1                1       1             0         NaN   
12                 1                1       1             0         NaN   
13                 1                1       1             0         NaN   
14                 1                1       1             0         NaN   
15                 1                1       1             0         NaN   
16                 1                1       1             0         NaN   
17                 1                1       1             0         NaN   
18                 1                1       1             0         NaN   
19                 1                1       1             0         NaN   

         NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                     NaN               1               1               1   
1                     NaN               0               0               0   
2                   Belém               1               1               1   
3   São Bernardo do Campo               1               1               1   
4                     NaN               0               0               0   
5                     NaN               1               1               1   
6                     NaN               0               0               0   
7                     NaN               0               0               0   
8                     NaN       

In [31]:
# OUTRA MANEIRA DE AJUSTAR
# dados1['RACA'] = dados1['RACA'].replace([0,1,2,3,4,5], ['Não Declarado','Branca','Preta','Parda','Amarela','Indígena'])

In [32]:
dados1["TP_LINGUA"] = dados1["TP_LINGUA"].replace({0:"Inglês", 1:"Espanhol"})

In [33]:
dados1["ESCOLA"] = dados1["ESCOLA"].replace({1:"não_respondeu", 2:"pública", 3:"privada", 4:"exterior"})

In [34]:
dados1["INTERNET"] = dados1["INTERNET"].replace({"A":"Não", "B":"Sim"})

## Análise Exploratória e Transformação dos dados

**Análise das idades**

In [35]:
dados1['IDADE'].value_counts()



IDADE
17    219313
18    147548
19     81236
16     59316
20     53009
       ...  
86         1
4          1
3          1
91         1
2          1
Name: count, Length: 80, dtype: int64

In [36]:
dados1['IDADE'].value_counts().sort_index()

IDADE
2     1
3     1
4     1
5     2
6     2
     ..
80    2
81    3
82    2
86    1
91    1
Name: count, Length: 80, dtype: int64

In [37]:
# Para ver todas as linhas, primeiro transformei num Dataframe:
idade = pd.DataFrame(dados1['IDADE'].value_counts().sort_index())

In [38]:
# Configurando o Pandas para observar 80 linhas
pd.set_option('display.max_rows', 80)

In [39]:
idade.head(80)

count
IDADE        
2           1
3           1
4           1
5           2
6           2
7           3
8           6
12          6
13         29
14        355
15      11583
16      59316
17     219313
18     147548
19      81236
20      53009
21      37242
22      29489
23      23531
24      18851
25      15103
26      12465
27      10367
28       8986
29       7650
30       6967
31       6331
32       5685
33       5175
34       4795
35       4340
36       4109
37       4105
38       3767
39       3299
40       3058
41       2638
42       2462
43       2307
44       2069
45       1939
46       1718
47       1578
48       1496
49       1304
50       1167
51       1014
52        933
53        841
54        727
55        652
56        530
57        497
58        388
59        325
60        254
61        236
62        176
63        144
64        139
65        115
66         81
67         72
68         53
69         43
70         40
71         25
72         17
73         16
74         13
75         10
76          8
77          3
78          3
79          4
80          2
81          3
82          2
86          1
91          1

Alguns valores estranhos e curiosos: idades menores que 12 anos e maiores de 80 anos.

In [40]:
menores_12 = dados1.query('IDADE < 12')['NO_MUNICIPIO_RESIDENCIA'].value_counts()

In [41]:
# Inscrições com idade abaixo de 12 anos serão excluídos devido a alta probabilidade de erro.
menores_12

NO_MUNICIPIO_RESIDENCIA
São Paulo                 5
Tambaú                    1
Ribeirão Preto            1
Bálsamo                   1
São Bernardo do Campo     1
Embu das Artes            1
Queiroz                   1
Santana de Parnaíba       1
Vargem Grande Paulista    1
Auriflama                 1
Araçatuba                 1
Campinas                  1
Name: count, dtype: int64

In [42]:
dados_maiores_11 = dados1.loc[dados1.IDADE > 11]

In [43]:
dados_maiores_11.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0                 1                1  não_respondeu             0         NaN   
1                 1                1  não_respondeu             0         NaN   
2                 1                2        privada             0  15567761.0   
3                 1                2        pública             0  35904958.0   
4                 1                1  não_respondeu             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  \
0               1    564.6    585.8    592.9    600.2    Inglês   
1               0      NaN      NaN      NaN      NaN  Espanhol   
2               1    576.2    641.1    634.3    731.8    Inglês   
3               1    402.2    424.8    509.6    393.2  Espanhol   
4               0      NaN      NaN      NaN      NaN  Espanhol   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0                1.0  160.0  200.0  180.0  200.0  200.0         940.0      Sim  
1                NaN    NaN    NaN    NaN    NaN    NaN           NaN      Não  
2                1.0  160.0  120.0  200.0  200.0  200.0         880.0      Não  
3                1.0   80.0  160.0  120.0  100.0  100.0         560.0      Sim  
4                NaN    NaN    NaN    NaN    NaN    NaN           NaN      Sim

In [44]:
dados_maiores_11['IDADE'].value_counts().sort_index()

IDADE
12         6
13        29
14       355
15     11583
16     59316
17    219313
18    147548
19     81236
20     53009
21     37242
22     29489
23     23531
24     18851
25     15103
26     12465
27     10367
28      8986
29      7650
30      6967
31      6331
32      5685
33      5175
34      4795
35      4340
36      4109
37      4105
38      3767
39      3299
40      3058
41      2638
42      2462
43      2307
44      2069
45      1939
46      1718
47      1578
48      1496
49      1304
50      1167
51      1014
52       933
53       841
54       727
55       652
56       530
57       497
58       388
59       325
60       254
61       236
62       176
63       144
64       139
65       115
66        81
67        72
68        53
69        43
70        40
71        25
72        17
73        16
74        13
75        10
76         8
77         3
78         3
79         4
80         2
81         3
82         2
86         1
91         1
Name: count, dtype: int64

**Treineiros**

In [45]:
dados_maiores_11.query('IN_TREINEIRO == 1')['IN_TREINEIRO'].value_counts()

IN_TREINEIRO
1    93988
Name: count, dtype: int64

In [46]:
treineiros = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 1]

In [47]:
treineiros.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
20      1,90E+11                  Santos     15    F                1   parda   
21      1,90E+11                Panorama     16    F                1   parda   
106     1,90E+11               São Paulo     16    F                1  branca   
109     1,90E+11        Águas de Lindóia     16    M                1  branca   
159     1,90E+11        Águas de Lindóia     15    M                1  branca   

     TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  \
20                  1                3  não_respondeu             1   
21                  1                3  não_respondeu             1   
106                 1                3  não_respondeu             1   
109                 1                3  não_respondeu             1   
159                 1                3  não_respondeu             1   

     CO_ESCOLA NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
20         NaN              NaN               1               1   
21         NaN              NaN               0               0   
106        NaN              NaN               1               1   
109        NaN              NaN               0               0   
159        NaN              NaN               0               0   

     TP_PRESENCA_LC  TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT  \
20                1               1    455.4    506.7    480.4    573.2   
21                0               0      NaN      NaN      NaN      NaN   
106               1               1    424.5    517.7    489.6    532.6   
109               0               0      NaN      NaN      NaN      NaN   
159               0               0      NaN      NaN      NaN      NaN   

    TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
20     Inglês                1.0  140.0  120.0  120.0  100.0   80.0   
21   Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
106    Inglês                1.0  120.0  120.0  120.0  100.0   80.0   
109  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
159    Inglês                NaN    NaN    NaN    NaN    NaN    NaN   

     NOTA_REDACAO INTERNET  
20          560.0      Sim  
21            NaN      Não  
106         540.0      Sim  
109           NaN      Sim  
159           NaN      Sim

In [48]:
treineiros.to_csv('treineiros_enem_2019_sp.csv', encoding = 'iso-8859-1', index = False)

In [49]:
# Separação entre os treineiros e os vestibulandos
vestibulandos = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 0]

In [50]:
vestibulandos.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0                 1                1  não_respondeu             0         NaN   
1                 1                1  não_respondeu             0         NaN   
2                 1                2        privada             0  15567761.0   
3                 1                2        pública             0  35904958.0   
4                 1                1  não_respondeu             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  \
0               1    564.6    585.8    592.9    600.2    Inglês   
1               0      NaN      NaN      NaN      NaN  Espanhol   
2               1    576.2    641.1    634.3    731.8    Inglês   
3               1    402.2    424.8    509.6    393.2  Espanhol   
4               0      NaN      NaN      NaN      NaN  Espanhol   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0                1.0  160.0  200.0  180.0  200.0  200.0         940.0      Sim  
1                NaN    NaN    NaN    NaN    NaN    NaN           NaN      Não  
2                1.0  160.0  120.0  200.0  200.0  200.0         880.0      Não  
3                1.0   80.0  160.0  120.0  100.0  100.0         560.0      Sim  
4                NaN    NaN    NaN    NaN    NaN    NaN           NaN      Sim

In [51]:
vestibulandos.shape

(719768, 29)

**Análise das presenças**

In [52]:
vestibulandos['TP_PRESENCA_CN'].value_counts()

TP_PRESENCA_CN
1    500805
0    218693
2       270
Name: count, dtype: int64

In [53]:
vestibulandos['TP_PRESENCA_CH'].value_counts()

TP_PRESENCA_CH
1    533142
0    185969
2       657
Name: count, dtype: int64

In [54]:
vestibulandos['TP_PRESENCA_LC'].value_counts()

TP_PRESENCA_LC
1    533142
0    185969
2       657
Name: count, dtype: int64

In [55]:
vestibulandos['TP_PRESENCA_MT'].value_counts()

TP_PRESENCA_MT
1    500805
0    218693
2       270
Name: count, dtype: int64

RESUMO DA PRESENÇA NAS PROVAS:

Ciências da Natureza: 500805 presentes, 218693 faltaram e 270 eliminados.

Ciências Humanas: 533142 presentes, 185969 faltaram e 657 eliminados.

Linguagens e Códigos: 533142 presentes, 185969 faltaram e 657 eliminados.

Matemática: 500805 presentes, 218693 faltaram e 270 eliminados.

In [56]:
vestibulandos['TP_STATUS_REDACAO'].value_counts().sort_index()

TP_STATUS_REDACAO
1.0    519763
2.0       913
3.0      1417
4.0      5513
6.0      3626
7.0       543
8.0       828
9.0       539
Name: count, dtype: int64

**Valores Ausentes**

In [57]:
vestibulandos['IDADE'].isnull().sum()

0

In [58]:
provas = ['NOTA_REDACAO','NOTA_CN','NOTA_MT','NOTA_LC','NOTA_CH']

In [59]:
# Registros NAN (nulo) nas notas
vestibulandos[provas].isnull().sum()

NOTA_REDACAO    186626
NOTA_CN         218963
NOTA_MT         218963
NOTA_LC         186626
NOTA_CH         186626
dtype: int64

In [60]:
enem_classificados = vestibulandos.loc[vestibulandos.TP_PRESENCA_CH == 1]
enem_classificados = enem_classificados.loc[enem_classificados.TP_PRESENCA_MT == 1]

In [61]:
enem_classificados['TP_PRESENCA_MT'].value_counts()

TP_PRESENCA_MT
1    499940
Name: count, dtype: int64

In [62]:
enem_classificados['TP_PRESENCA_CH'].value_counts()

TP_PRESENCA_CH
1    499940
Name: count, dtype: int64

In [63]:
enem_classificados['TP_PRESENCA_CN'].value_counts()

TP_PRESENCA_CN
1    499940
Name: count, dtype: int64

In [64]:
enem_classificados['TP_PRESENCA_LC'].value_counts()

TP_PRESENCA_LC
1    499940
Name: count, dtype: int64

In [65]:
enem_classificados[provas].isnull().sum()

NOTA_REDACAO    0
NOTA_CN         0
NOTA_MT         0
NOTA_LC         0
NOTA_CH         0
dtype: int64

** Notas zeros na Redação**

In [66]:
from collections import Counter
dict(Counter(enem_classificados['NOTA_REDACAO']))

{940.0: 4688,
 880.0: 7676,
 560.0: 35741,
 580.0: 33062,
 540.0: 25538,
 900.0: 5921,
 520.0: 24551,
 500.0: 17102,
 680.0: 19520,
 660.0: 18451,
 760.0: 12859,
 860.0: 6502,
 480.0: 19617,
 640.0: 27623,
 720.0: 15214,
 600.0: 37197,
 820.0: 7810,
 340.0: 5464,
 0.0: 9181,
 920.0: 6540,
 780.0: 9236,
 400.0: 7232,
 740.0: 10931,
 320.0: 8113,
 840.0: 9047,
 800.0: 10677,
 700.0: 13502,
 460.0: 13045,
 360.0: 7023,
 620.0: 28059,
 980.0: 978,
 440.0: 11950,
 420.0: 6555,
 380.0: 4453,
 280.0: 4734,
 300.0: 5418,
 960.0: 3057,
 200.0: 312,
 260.0: 2634,
 180.0: 105,
 240.0: 1841,
 220.0: 664,
 40.0: 1,
 140.0: 20,
 160.0: 65,
 120.0: 16,
 100.0: 5,
 60.0: 4,
 1000.0: 5,
 80.0: 1}

In [67]:
enem_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

NOTA_REDACAO
0.0    9181
Name: count, dtype: int64

In [68]:
enem_classificados.query('COMP1 == 0')['COMP1'].value_counts()

COMP1
0.0    9188
Name: count, dtype: int64

In [69]:
enem_classificados.query('COMP2 == 0')['COMP2'].value_counts()

COMP2
0.0    9181
Name: count, dtype: int64

In [70]:
enem_classificados.query('COMP3 == 0')['COMP3'].value_counts()

COMP3
0.0    9197
Name: count, dtype: int64

In [71]:
enem_classificados.query('COMP4 == 0')['COMP4'].value_counts()

COMP4
0.0    9194
Name: count, dtype: int64

In [72]:
enem_classificados.query('COMP5 == 0')['COMP5'].value_counts()

COMP5
0.0    73330
Name: count, dtype: int64

NOTAS ZEROS EM REDAÇÃO:

REDAÇÃO (final): 9181 notas zeros devido a não cumprimento das regras ou tirou zero em todos os cinco componentes.

COMPONENTE 1(Demonstrar domínio da modalidade escrita formal da Língua Portuguesa.): 9188 notas zeros somente neste item.

COMPONENTE 2 (Compreender a proposta de redação e aplicar conceitos das várias áreas de conhecimento para desenvolver o tema, dentro dos limites estruturais do texto dissertativo-argumentativo em prosa): 9181.

COMPONENTE 3 (Selecionar, relacionar, organizar e interpretar informações, fatos, opiniões e argumentos em defesa de um ponto de vista): 9197 notas zeros somente neste item.

COMPONENTE 4 (Demonstrar conhecimento dos mecanismos linguísticos necessários para a construção da argumentação.): 9194 notas zeros somente neste item.

COMPONENTE 5 (Elaborar proposta de intervenção para o problema abordado, respeitando os direitos humanos): 73330 notas zeros somente neste item.


In [73]:
enem_classificados.query('NOTA_CH == 0')['NOTA_CH'].value_counts()

NOTA_CH
0.0    343
Name: count, dtype: int64

In [74]:
enem_classificados.query('NOTA_CN == 0')['NOTA_CN'].value_counts()

NOTA_CN
0.0    36
Name: count, dtype: int64

In [75]:
enem_classificados.query('NOTA_LC == 0')['NOTA_LC'].value_counts()

NOTA_LC
0.0    135
Name: count, dtype: int64

In [76]:
enem_classificados.query('NOTA_MT == 0')['NOTA_MT'].value_counts()

NOTA_MT
0.0    56
Name: count, dtype: int64

In [77]:
enem_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

NOTA_REDACAO
0.0    9181
Name: count, dtype: int64

NOTAS ZEROS DOS INSCRITOS CONCORRENTES AO VESTIBULAR:

Matemática = 56

Ciências da Natureza = 36

Linguagens e Códigos = 135

Ciências Humanas = 343

Redação = 9181


In [78]:
enem_classificados.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
5     1,90E+11              Embu-Guaçu     17    F                1   parda   
8     1,90E+11               Guarulhos     23    F                1  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0                 1                1  não_respondeu             0         NaN   
2                 1                2        privada             0  15567761.0   
3                 1                2        pública             0  35904958.0   
5                 1                2        pública             0         NaN   
8                 1                1  não_respondeu             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
5                    NaN               1               1               1   
8                    NaN               1               1               1   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  \
0               1    564.6    585.8    592.9    600.2    Inglês   
2               1    576.2    641.1    634.3    731.8    Inglês   
3               1    402.2    424.8    509.6    393.2  Espanhol   
5               1    399.6    426.8    487.2    419.2    Inglês   
8               1    625.7    646.5    629.3    796.1    Inglês   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0                1.0  160.0  200.0  180.0  200.0  200.0         940.0      Sim  
2                1.0  160.0  120.0  200.0  200.0  200.0         880.0      Não  
3                1.0   80.0  160.0  120.0  100.0  100.0         560.0      Sim  
5                1.0  120.0  120.0  120.0  120.0   80.0         560.0      Sim  
8                1.0  160.0  200.0  180.0  180.0  160.0         880.0      Sim

In [79]:
enem_classificados.shape

(499940, 29)

## **Salvando (Exportando) o Dataframe Tratado**

In [81]:
enem_classificados.to_csv('enem_2019_tratado.csv', encoding = 'iso-8859-1', index = False)